minisom.py

```diff
@@ -390,15 +391,23 @@ def activation_response(self, data):
         for x in data:
             a[self.winner(x)] += 1
         return a
+    
+    def dist_mat(self, data):
+        """
+        calculate distance matrix: dist[i,j]: i: the i-th data, j: the j-th node
+        """
+        D = np.array(data)
+        C = self._weights.reshape(-1, self._weights.shape[2])
+        dd = (D ** 2).sum(axis=1, keepdims=True)
+        cc = (C ** 2).sum(axis=1, keepdims=True)
+        dc = D @ C.T
+        return np.sqrt(-2 * dc + dd + cc.T)
```

Comment:
wouldn't it be simpler to compute the distances as `(np.sum((data[None,:] - self._weights[:, None])**2, -1)**0.5)`?